### ISSUE TITLE
Add possibility for AssemblyMetadata collection in CreateAssemblyInfo

### ISSUE THREAD

**Issue by WebDucer** [cake-build/cake#2003]
There is no possibility to add AsselmbyMetadata attribute to the assembly info file, because this needs two string paramters. The CustomAttribute allows only one. It would be nice, to have something similar to `CustomAttributes` property and `.AddCustomAttribute` extension method.

```csharp
var version = GitVersion();
CreateAssemblyInfo("./AssemblyInfo.cs", new AssemblyInfoSettings {
    Product = "SampleProject",
    Version = version.AssemblySemVer,
    FileVersion = version.AssemblySemVer,
    InformationalVersion = version.InformationalVersion,
    Copyright = string.Format("Copyright (c) Company 2018 - {0}", DateTime.Now.Year)
}
.AddAssemblyMetadataAttribute("git_sha", version.Sha)
.AddAssemblyMetadataAttribute("semantic_version", version.SemVer)
.AddAssemblyMetadataAttribute("build_date", DateTime.Now.ToString("yyy-MM-dd HH:mm")
);
```

Should produce this:

```csharp
using System.Reflection;
[assembly:AssemblyProduct("SampleProject")]
[assembly: AssemblyVersion("0.1.0.0")]
[assembly: AssemblyInformationalVersion("0.1.0-alpha.238+Branch.feature/Info_page.Sha.c73ac3b7ca0e00ae29ec769dc4afe770a38d6d1e")]
[assembly: AssemblyFileVersion("0.1.0.0")]
[assembly:AssemblyCopyright("Copyright (c) Company 2018 - 2018")]

[assembly: AssemblyMetadata("semantic_version", "0.1.0-alpha.238")]
[assembly: AssemblyMetadata("git_sha", "c73ac3b7ca0e00ae29ec769dc4afe770a38d6d1e")]
[assembly: AssemblyMetadata("build_date", "2018-01-28 22:35")]
```

**Comment by WebDucer** [comment]
Is there any chance to get this meta data possibility in one of the next releases?

**Comment by devlead** [comment]
@WebDucer we're currently focusing on getting 0.26.0 out the door which will be a big move from .NET Core 1.x to .NET Core 2.x. runtime & sdk.
Once that's out which shouldn't be to long, just finishing the last bits on the PR, then we'll start triaging PRs and issues for the 0.27.0 release which our goal is to release fairly quickly after 0.26.0.

**Comment by WebDucer** [comment]
@devlead Thanks. This sounds (.Net Core 2.0) realy good.

**Comment by mholo65** [comment]
Closed #2004
